docs/devcontainer-example/devcontainer.json

```diff
@@ -0,0 +1,74 @@
+{
+  "build": { "dockerfile": "Dockerfile.llvm-project" },
+  "customizations": {
+    "vscode": {
+      "settings": {
+        "terminal.integrated.defaultProfile.linux": "bash",
+        "cmake.mergedCompileCommands": "${workspaceFolder}/compile_commands.json",
+        "cmakeExplorer.suiteDelimiter": ".",
+        "cmakeExplorer.debugConfig": "(gdb) Launch",
+        "editor.formatOnSave": true,
+        "[cpp]": {
+          "editor.defaultFormatter": "xaver.clang-format",
+          "editor.tabSize": 2
+        },
+        "clangd.path": "/usr/bin/clangd-15",
+        "clangd.arguments": [
+          "--background-index",
+          "-header-insertion=never",
+          "-j=4"
+        ],
+        "cmake.buildDirectory": "${workspaceFolder}/build",
+        "cmake.generator": "Ninja",
+        "cmake.defaultVariants": {
+          "buildType": {
+            "default": "reldeb",
+            "description": "The build type.",
+            "choices": {
+              "debug": {
+                "short": "Debug",
+                "long": "Disable optimizations - include debug information.",
+                "buildType": "Debug"
+              },
+              "release": {
+                "short": "Release",
+                "long": "Optimize for speed - exclude debug information.",
+                "buildType": "Release"
+              },
+              "minsize": {
+                "short": "MinSizeRel",
+                "long": "Optimize for smallest binary size - exclude debug information.",
+                "buildType": "MinSizeRel"
+              },
+              "reldeb": {
+                "short": "RelWithDebInfo",
+                "long": "Optimize for speed - include debug information.",
+                "buildType": "RelWithDebInfo"
+              }
+            }
+          }
+        }
+      },
+      "extensions": [
+        "llvm-vs-code-extensions.vscode-clangd",
+        "vadimcn.vscode-lldb",
+        "ms-vscode.cmake-tools",
+        "fredericbonnet.cmake-test-adapter",
+        "twxs.cmake",
+        "xaver.clang-format",
+        "notskm.clang-tidy",
+        "ms-python.python",
+        "ms-python.vscode-pylance",
+        "eamodio.gitlens",
+        "llvm-vs-code-extensions.vscode-mlir",
+        "cschlosser.doxdocgen",
+        "chneau.resource-monitor",
+        "spmeesseman.vscode-taskexplorer",
+        "edonet.vscode-command-runner"
+      ]
+    }
+  },
+  "workspaceFolder": "${localWorkspaceFolder}",
+  "workspaceMount": "source=${localWorkspaceFolder},target=${localWorkspaceFolder},type=bind,consistency=cached",
+  "postAttachCommand": "ONNX_ROOT=third_party/onnx && cd ${ONNX_ROOT} && python3 -m pip install . && cd ../.."
```

Comment:
I'm getting an error:
```
ERROR: Directory '.' is not installable. Neither 'setup.py' nor 'pyproject.toml' found.
[29678 ms] postAttachCommand failed with exit code 1. Skipping any further user-provided commands.
```
I think we need a git submodule sync when using a fresh clone.

Reply:
Yes, there is this line in the docs `To use this setup you first clone onnx-mlir and all submodules.` I willprobably makes this statement bold...

Reply:
Ah my bad! Then all good

Reply:
Would this be `git {clone|checkout} --recurse-submodules`?

Reply:
Yes. I think I used `git clone {onnx-mlir-url}` and `git submodule update --init --recursive`

Reply:
In https://github.com/onnx/onnx-mlir/blob/main/docs/BuildOnLinuxOSX.md they say `git clone --recursive https://github.com/onnx/onnx-mlir.git`